### DISCUSSION TITLE
accessing iframe context within a shadowroot

### DISCUSSION THREAD

**Discussion by pjacko** [cefsharp/CefSharp.Dom#49] [General]
Hi there,
I need to access an iframe's windowContext and documentContext within a (open) shadowRoot and am unsure how to do it (?)
From a javascript point of view it would look like this

> var ifr = document.querySelector('legacy-miniapp').shadowRoot.querySelector('iframe');
> var ifrEle = ifr.windowContext.$('#eleId');

**Comment by amaitland** [comment]
There are likely a number of ways. Generally I'd start with seeing how you'd achieve something in `puppereer sharp` for an example and go from there. The most efficient is likely just to execute your existing javascript. Something like

```c#
var handle = await devToolsContext.EvaluateFunctionHandleAsync("document.querySelector('legacy-miniapp').shadowRoot.querySelector('iframe');");
var iFrame = handle.ToDomHandle<HtmlInlineFrameElement>();
```

https://github.com/cefsharp/CefSharp.Dom/blob/dc7f2928e53afcf1061e3026439362ef269b7a04/lib/PuppeteerSharp.Tests/ElementHandleTests/ContentFrameTests.cs#L24
https://github.com/cefsharp/CefSharp.Dom/blob/dc7f2928e53afcf1061e3026439362ef269b7a04/lib/PuppeteerSharp/JSHandle.cs#L262

**Comment by pjacko** [comment]
Thanks alot Alex, 
I'll give it a whirl!
